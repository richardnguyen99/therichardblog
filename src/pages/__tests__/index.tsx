/**
 * @jest-environment jsdom
 */

import * as React from "react";
import { render } from "@testing-library/react";

import IndexPage from "../index";

const customRender = (ui) => {
  return render(<>{ui}</>);
};

const mockUseLocationValue = {
  pathname: "/",
  search: "",
  hash: "",
  state: null,
};

const mockUseStaticQueryValue = {
  site: {
    siteMetadata: {
      defaultTitle: "The Richard's blog",
      titleTemplate: "%s - The Richard's blog",
      defaultDescription: "A digital notebook from Richard",
      siteUrl: "https://therichardblog.com",
      defaultImage: "src/images/icon.png",
      twitterUsername: "@richardnguyenmh",
      github: "https://github.com/richardnguyen99",
      linkedin: "https://www.linkedin.com/in/richardmhnguyen/",
    },
  },
};

jest.mock("@gatsbyjs/reach-router", () => ({
  ...(jest.requireActual("@gatsbyjs/reach-router") as Record<string, unknown>),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  }),
}));

jest.mock("gatsby", () => ({
  ...(jest.requireActual("gatsby") as Record<string, unknown>),
  useStaticQuery: jest.fn().mockImplementation(() => {
    return mockUseStaticQueryValue;
  }),
  graphql: jest.fn(),
}));

describe("<Index />", () => {

  it("should display the page heading", () => {
    const { getByText } = customRender(<IndexPage />);

    expect(getByText("Hello, World")).toBeInTheDocument();
  });
});
