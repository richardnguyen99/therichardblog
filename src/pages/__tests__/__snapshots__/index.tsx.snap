// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should match snapshot 1`] = `
<div>
  <main>
    <div>
      <nav
        class="sc-gsTEea dgGraA"
      >
        <div
          class="sc-dlfnuX jPvAmc"
        >
          <a
            aria-current="page"
            class="sc-jSgvzq sc-gKseQn lmTSet eEosdZ"
            href="/"
          >
            The Richard's blog
          </a>
          <div
            class="sc-eCstlR lfkxJZ"
          >
            <a
              class="sc-jSgvzq lmTSet"
              href="/about"
            >
              About
            </a>
            <a
              class="sc-jSgvzq lmTSet"
              href="/post"
            >
              Post
            </a>
          </div>
          <div
            class="sc-eCstlR eRDLs"
          >
            <a
              class="sc-jSgvzq lmTSet"
              href="https://github.com/richardnguyen99/therichardblog"
            >
              Github
            </a>
            <span
              class="sc-jSgvzq lmTSet"
              href=""
              to=""
            >
              <div
                class="sc-iBPTik bUenXA"
              >
                <div
                  class="sc-kEjbQP zinsM"
                >
                  <div
                    class="sc-iqHYmW dEJZyq"
                    style="opacity: 0;"
                  >
                    More
                  </div>
                  <div>
                    <button
                      class="sc-pGacB kyGELd"
                      style="padding: 0px;"
                    >
                      <svg
                        fill="white"
                        fill-rule="evenodd"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zM1.5 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zm13 0a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <ul
                  class="sc-fubCzh GdjJR southwest"
                  style="opacity: 0; display: false;"
                >
                  <li
                    class="sc-jrAFXE hJCAtD"
                  >
                    Tags
                  </li>
                  <li
                    class="sc-jrAFXE hJCAtD"
                  >
                    Tags
                  </li>
                  <li
                    class="sc-jrAFXE hJCAtD"
                  >
                    Tags
                  </li>
                </ul>
              </div>
            </span>
          </div>
        </div>
        <div
          class="sc-hKgJUU DNWkX"
          style="display: none; top: 76px; width: 0px; left: 0px;"
        />
      </nav>
      <div
        id="content"
      >
        <section
          class="sc-hHfuMS jlgQYN"
        >
          <h1
            class="sc-dmlqKv GgidW"
          >
            The latest from Richard.
          </h1>
        </section>
      </div>
      <footer
        class="sc-crrszt bRxQMy"
      >
        <div
          class="sc-dQoVA uTcuO"
        >
          <div
            class="sc-bqyKOL fnObYe"
          >
            <div
              class="sc-kstqJO khiLwQ"
            >
              The Richard's blog
            </div>
            <div
              class="sc-hBEYId bEpSCn"
            >
              <div
                class="sc-fodVek"
              >
                <h4
                  class="sc-fFucqa fUsbUe"
                >
                  Connection
                </h4>
                <ul
                  class="sc-bkzYnD klRfed"
                >
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="sc-fodVek"
              >
                <h4
                  class="sc-fFucqa fUsbUe"
                >
                  Connection
                </h4>
                <ul
                  class="sc-bkzYnD klRfed"
                >
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="sc-fodVek"
              >
                <h4
                  class="sc-fFucqa fUsbUe"
                >
                  Connection
                </h4>
                <ul
                  class="sc-bkzYnD klRfed"
                >
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="sc-fodVek"
              >
                <h4
                  class="sc-fFucqa fUsbUe"
                >
                  Connection
                </h4>
                <ul
                  class="sc-bkzYnD klRfed"
                >
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                  <li
                    class="sc-idOiZg iHieYo"
                  >
                    <a
                      href="#"
                    >
                      Facebook
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="sc-dIUeWJ kjSuug"
          >
            <span>
              <img
                alt="Made by Richard with love"
                src="https://img.shields.io/static/v1?label=Made by Richard&message=with â™¥&color=red"
              />
            </span>
            <span>
              <a
                href="https://github.com/richardnguyen99/therichardblog/"
              >
                Github repo
              </a>
            </span>
          </div>
        </div>
      </footer>
    </div>
  </main>
</div>
`;
